server/services.py

```
@@ -0,0 +1,24 @@
+
+async def chat_service(model: str, messages: list[Message]) -> list:
+    try:
+        completion = client.chat.completions.create(
+            model=model,
+            messages=[msg.model_dump() for msg in messages]
+        )
+        assistant_content = completion.choices[0].message.content
+        response_messages = messages + [Message(role=Role.ASSISTANT, content=assistant_content)]
+        return response_messages
+    except openai.AuthenticationError:
+        raise AuthenticationError("Invalid OpenAI API Key")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)